### PR TITLE
Use derivable icons for image scaling to enable lossless icon scaling…

### PR DESCRIPTION
**Pull request by weisJ**
… where possible.

Implementing derivable icon types enabled lossless icon scaling for non `Image` based icons.

It's using some features (and needs some bug fixes) from the current darklaf pre-release.  That's why it's using the `latest.integration` version.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -9,18 +9,25 @@ plugins {
...
     implementation 'org.scribe:scribe:1.3.0'
     implementation 'org.hsqldb:hsqldb:2.4.1'
     implementation 'com.google.code.gson:gson:2.8.6'
     implementation 'com.squareup.okhttp3:okhttp:4.3.1'
-    implementation 'com.github.weisj:darklaf-core:2.2.3'
-    implementation 'com.github.weisj:darklaf-theme:2.2.3'
-    implementation 'com.github.weisj:darklaf-property-loader:2.2.3'
+    implementation 'com.github.weisj:darklaf-core:latest.integration'
+    implementation 'com.github.weisj:darklaf-theme:latest.integration'
+    implementation 'com.github.weisj:darklaf-property-loader:latest.integration'
```
Comment (tychobrailleur) [inline build.gradle]: Are the changes here required for this PR to work?  I'd rather keep dependencies on fixed released versions.

**Review by tychobrailleur — COMMENTED** [review]
Thanks for this!  The changes make good sense, other than the changes to the build file.

File: build.gradle (hunk 1 above)
Comment (weisJ) [reply build.gradle]: They are necessary for now. I am going to release the next version of darklaf later today wich include the necessary fixes.

Comment (tychobrailleur) [reply build.gradle]: Apologies, I only saw your note to that end just now – perfect, I am happy to wait for the release, no rush.

**Comment by weisJ** [comment]
I have adjusted the PR to use the new darklaf release. Removed the necessity for the snapshot versions.

**Pull request merged**